src/test/kotlin/com/cultureamp/eventsourcing/UpcastTest.kt

```diff
@@ -0,0 +1,109 @@
+import com.cultureamp.eventsourcing.*
+import com.cultureamp.eventsourcing.example.*
+import com.cultureamp.eventsourcing.sample.StandardEventMetadata
+import io.kotest.core.spec.style.DescribeSpec
+import io.kotest.matchers.shouldBe
+import org.jetbrains.exposed.sql.Database
+import org.jetbrains.exposed.sql.SchemaUtils
+import org.jetbrains.exposed.sql.StdOutSqlLogger
+import org.jetbrains.exposed.sql.addLogger
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.joda.time.DateTime
+import java.util.*
+
+class UpcastTest : DescribeSpec({
+    val db = Database.connect(url = "jdbc:h2:mem:test;MODE=MySQL;DB_CLOSE_DELAY=-1;", driver = "org.h2.Driver")
```

Comment:
pet peeve: we use Postgres not MySQL
```suggestion
    val db = Database.connect(url = "jdbc:h2:mem:test;MODE= PostgreSQL;DB_CLOSE_DELAY=-1;", driver = "org.h2.Driver")
```
Doesn't make much difference though